src/topology.jl

```diff
@@ -190,3 +190,189 @@ function Cart_sub(comm::Comm, remain_dims)
+
+# External links
+$(_doc_external("MPI_Dist_graph_create_adjacent"))
+"""
+function Dist_graph_create_adjacent(comm::Comm, sources::Vector{Cint}, destinations::Vector{Cint}; source_weights::Union{Vector{Cint}, Ptr{Cvoid}}=UNWEIGHTED[], destination_weights::Union{Vector{Cint}, Ptr{Cvoid}}=UNWEIGHTED[], reorder=false, infokws...)
+    graph_comm = Comm()
+    # int MPI_Dist_graph_create_adjacent(MPI_Comm comm_old,
+    #       int indegree, const int sources[],
+    #       const int sourceweights[],
+    #       int outdegree, const int destinations[],
+    #       const int destweights[],
+    #       MPI_Info info, int reorder, MPI_Comm *comm_dist_graph)
+    @mpichk ccall((:MPI_Dist_graph_create_adjacent, libmpi), Cint,
+         (MPI_Comm, Cint, Ptr{Cint}, Ptr{Cint}, Cint, Ptr{Cint}, Ptr{Cint}, MPI_Info, Cint, Ptr{MPI_Comm}),
+         comm,length(sources),sources,source_weights,length(destinations),destinations,destination_weights,Info(infokws...),reorder,graph_comm)
+    return graph_comm
+end
+
+"""
+    graph_comm = Dist_graph_create(comm::Comm, sources::Vector{Cint}, degrees::Vector{Cint}, destinations::Vector{Cint}; weights::Union{Vector{Cint}, Ptr{Cvoid}}=UNWEIGHTED[], reorder=false, infokws...)
+
+Create a new communicator from a given directed graph topology, described by incoming and outgoing edges on an existing communicator.
+
+# External links
+$(_doc_external("MPI_Dist_graph_create"))
+"""
+function Dist_graph_create(comm::Comm, sources::Vector{Cint}, degrees::Vector{Cint}, destinations::Vector{Cint}; weights::Union{Vector{Cint}, Ptr{Cvoid}}=UNWEIGHTED[], reorder=false, infokws...)
+    graph_comm = Comm()
+    # int MPI_Dist_graph_create(MPI_Comm comm_old, int n, const int sources[],
+    #       const int degrees[], const int destinations[],
+    #       const int weights[],
+    #       MPI_Info info, int reorder, MPI_Comm *comm_dist_graph)
+    @mpichk ccall((:MPI_Dist_graph_create, libmpi), Cint,
+         (MPI_Comm, Cint, Ptr{Cint}, Ptr{Cint}, Ptr{Cint}, Ptr{Cint}, MPI_Info, Cint, Ptr{MPI_Comm}),
+         comm,length(sources),sources,degrees,destinations,weights,Info(infokws...),reorder,graph_comm)
+    return graph_comm
+end
+
+"""
+    indegree, outdegree, weighted = Dist_graph_neighbors_count(graph_comm::Comm)
+
+Return the number of in and out edges for the calling processes in a distributed graph topology and a flag indicating whether the distributed graph is weighted.
+
+# External links
+$(_doc_external("MPI_Dist_graph_neighbors_count"))
+"""
+function Dist_graph_neighbors_count(graph_comm::Comm)
+    indegree = Ref{Cint}()
+    outdegree = Ref{Cint}()
+    weighted = Ref{Cint}()
+    # int MPI_Dist_graph_neighbors_count(MPI_Comm comm, int *indegree, int *outdegree, int *weighted)
+    @mpichk ccall((:MPI_Dist_graph_neighbors_count, libmpi), Cint,
+        (MPI_Comm, Ptr{Cint}, Ptr{Cint}, Ptr{Cint}),
+        graph_comm,indegree,outdegree,weighted)
+    (indegree[], outdegree[], weighted[])
```

Comment:
```suggestion
    (indegree[], outdegree[], weighted[] != 0)
```
Might as well return a `Bool`